pandas/core/common.py

```diff
@@ -1403,6 +1403,10 @@ def _fill_zeros(result, x, y, name, fill):
                 if np.isinf(fill):
                     np.putmask(result,(signs<0) & mask, -fill)
 
+                if "floordiv" in name:  # (PR 9308)
+                    nan_mask = ((y == 0) & (x == 0)).ravel()
+                    np.putmask(result, nan_mask, np.nan)
+
```

Comment:
Yeah, good call on the profiling!

Raveling and reshaping is helping.  They're not causing any copies to be made which I confirmed by memory profiling.  

But actually, there was one thing causing an unnecessary copy to be made (independent of whether we ravel or not) which was using `.astype('float64')`.  I've changed it to `.astype('float', copy=False)`.    The reason for changing 'float64' to 'float' is that I don't see any reason to force float64; isn't it better to allow the computer architecture to decide between float32 and float64?

After this change, I tried time profiling again to verify that the raveling/reshaping is still helping.